using-timescaledb/continuous-aggregates.md

```diff
@@ -16,145 +16,189 @@ every refresh. Only new and/or invalidated data will be calculated.  Since this
-"eagerly" but can consume background worker resources while it is running.
-
-The `timescaledb.max_interval_per_job` parameter can also be useful, especially
-when a continuous aggregate is created on a hypertable that already contains
-data and the materialization job needs to catch up. It determines the maximum
-amount of data processed in a single job. If there is still more work to do to
-catch up to where we need to be based on the `refresh_lag` setting, another job
-will be spawned immediately following completion. This means that the part of the
-data processed by previous jobs will be available even before the
-materialization is fully caught up.
+### Advanced Usage
+
+When designing continuous aggregates, we wanted to avoid write
+amplification since this drastically affect insert rate for
+write-heavy time-series workloads. As a result, data is aggregated
+into a materialized hypertable (one for each continuous aggregate
+view) containing partial aggregates trailing the underlying hypertable
+(the table `conditions` in the example above).
+
+Each continuous aggregate has a *bucket width* that comes from the
+`time_bucket` call in the view definition. This is the granularity
+that is updated and typically new data is added to the last bucket. By
+not updating the last bucket of the view, we can avoid having to
+refresh the last bucket multiple times as new data comes in. Buckets
+older than that will be aggregated, but since new data does not
+(normally) arrive into those bucket, an aggegate for the bucket is
+computed only once, which avoid write amplification.
+
+However, when querying the continuous aggregate view (for example,
+`conditions_summary_daily`) you typically want to get a complete view
+of the data, including the data in the underlying table, so the view
+will get one part of the result from the materialized hypertable and
+one part from the underlying table.
+
+If you, however, want to just get the data from the materialized
+hypertable, and not include data from the underlying table, you can
+set the option `timescaledb.materialized_only` to `true` using `ALTER
+VIEW`:
+
+```sql
+ALTER VIEW conditions_summary_hourly SET (
+    timescaledb.materialized_only = true
+);
+```
+
+As noted above, the materialized hypertable lag behind the underlying
+table by one bucket width (parameter `bucket_width` in the
+`time_bucket` function of the view). However, you can change this lag
+to either increase it (or decrease it) by setting the value for the
+`timescaledb.refresh_lag` option of the view and the background worker
+will attempt to materialize everything before `refresh_lag` +
+`bucket_width`.
+
+For example, using `conditions_summary_hourly` above, the bucket width
+is 1 hour, so if we set the `refresh_lag` to 1 hour, the
+materialization will be 2 hours behind the current time.
+
+```sql
+ALTER VIEW conditions_summary_hourly SET (
+  timescaledb.refresh_lag = '1 hour'
```

Comment:
Here the user sets `refresh_lag` to 1 hour but in reality we do not honor this setting and instead the lag will be something else underneath-the-hood. In this case 2 hours, but could be something else depending on time_bucket.

I am pointing this out because there is such an obvious UX issue here and I am wondering if we can handle this better?